### PR TITLE
Add app and catalog

### PR DESCRIPTION
**Pull request by tomahawk28**

### CODE REVIEW

File: pkg/chartvalues/apiextensions_app_e2e.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+type APIExtensionsAppE2EConfigConfigMap struct {
+	ValuesYAML string
+}
+
+type APIExtensionsAppE2EConfigSecret struct {
+	ValuesYAML string
+}
+
+// NewAPIExtensionsAppE2E renders values required by
+// apiextensions-azure-config-e2e-chart.
```
Comment (rossf7) [inline pkg/chartvalues/apiextensions_app_e2e.go]: Chart name is wrong.

File: pkg/chartvalues/apiextensions_app_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
+		{
+			name: "case 1: invalid .Chart.Namespace",
+			config: newAPIExtensionsChartE2EConfigFromFilled(func(v *APIExtensionsChartE2EConfig) {
+				v.Chart.Namespace = ""
+			}),
+			errorMatcher: IsInvalidConfig,
+		},
```
Comment (rossf7) [inline pkg/chartvalues/apiextensions_app_e2e_test.go:160]: Tests need changing for the app chart. There is no tarball URL etc.

**Review by rossf7 — COMMENTED** [review]
Direction is good but there are some copy paste issues to fix.

File: pkg/chartvalues/apiextensions_app_e2e.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
+package chartvalues
+
+import (
+	"github.com/giantswarm/e2etemplates/internal/render"
```
Comment (rossf7) [inline pkg/chartvalues/apiextensions_app_e2e.go]: Minor but these should be in separate blocks.

File: pkg/chartvalues/apiextensions_app_e2e.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	AppOperator APIExtensionsAppE2EConfigAppOperator
+	ConfigMap   APIExtensionsAppE2EConfigConfigMap
+	Namespace   string
+	Secret      APIExtensionsAppE2EConfigSecret
+}
+
+type APIExtensionsAppE2EConfigApp struct {
+	Config    APIExtensionsAppE2EConfigAppConfig
+	Name      string
+	Namespace string
```
Comment (rossf7) [inline pkg/chartvalues/apiextensions_app_e2e.go:22]: I think the Version should also be configurable. We'll need that in the e2e tests to test updating apps.

**Review by rossf7 — APPROVED** [review]
LGTM but I think the app version also needs to be configurable. It would be good to get that sorted.

File: pkg/chartvalues/apiextensions_app_e2e.go (hunk 4 above)
Comment (tomahawk28) [reply pkg/chartvalues/apiextensions_app_e2e.go:22]: @rossf7 Which version do you mean? In label?
```
      labels:
        app: "prometheus"
        app-operator.giantswarm.io/version: "1.0.0"
```

Comment (tomahawk28) [reply pkg/chartvalues/apiextensions_app_e2e.go:22]: Ah, in spec. 
Sorry
```
   spec:
      catalog: "giantswarm"
      name: “prometheus”
      namespace: “monitoring”
      version: "3.2.1"
```

Comment (rossf7) [reply pkg/chartvalues/apiextensions_app_e2e.go:22]: No worries but yes we need the version to generate the tarball URLs. So we need to test that.

**Comment by tomahawk28** [comment]
@ross PTAL :)

**Review by rossf7 — APPROVED** [review]
Nice, thx!

**Pull request merged**